### ISSUE TITLE
How can I use the migration plugin I've built locally?

### ISSUE THREAD

**Issue by IngoStrauch2020** [mayope/keycloakmigration#86]
Is there a way to modify the plugin, build and publish it locally, and then use the local copy in a second project?

I tried with "publishToMavenLocal" but that is not picked up by the other project when I use my local SNAPSHOT version. I tried to add `mavenLocal()` to `repositories`, `buildscript.repositories` and `pluginManagement.repositories` but maybe something has to be changed for "publishToMavenLocal", I dunno.

Any pointers into the right direction?

**Comment by klg71** [comment]
Gradle Plugins aren't easy to use locally. You could try to copy the whole plugin in a `buildPlugins` folder of your target project. Then you can use `includeBuild("buildPlugins")` in your settings.gradle and then try to include it in the gradle `plugins{}` block.
I have not tried this method yet for the keycloakmigration plugin but it worked for other gradle plugins I work with.

**Comment by IngoStrauch2020** [comment]
It seems I got it working by using this in my `settings.gradle`

```
includeBuild("C:/dev/keycloakmigration") {
    dependencySubstitution {
        substitute(module("de.klg71.keycloakmigrationplugin:de.klg71.keycloakmigrationplugin.gradle.plugin"))
                .using(project(":plugin"))
        substitute(module("de.klg71:keycloakmigration"))
                .using(project(":"))
        substitute(module("de.klg71:keycloakapi"))
                .using(project(":keycloakapi"))
    }
}
```

It required several other changes to get a working combination of target JVM, kotlin compiler and gradle versions between both projects.
But now I can test my local changes.
Thanks for the pointer in the right direction. :-)

**Comment by klg71** [comment]
Released with version 0.0.72. Could you check if it works @IngoStrauch2020 ? :)

**Comment by IngoStrauch2020** [comment]
Cool, thanks for merging!

And yes, 0.2.72 works as expected.

I will create an additional PR with some documentation later.
